File: libgomp/testsuite/libgomp.oacc-fortran/pr70828.f90
! Subarrays on data construct: explicit-shape array.

program test
  integer, parameter :: n = 100
  integer i, data(n)

  data(:) = 0

  !$acc data copy(data(5:n-10))
  !$acc parallel loop
  do i = 10, n - 10
     data(i) = i
  end do
  !$acc end parallel loop
  !$acc end data

  do i = 1, n
     if ((i < 10 .or. i > n-10)) then
        if ((data(i) .ne. 0)) call abort
     else if (data(i) .ne. i) then
        call abort
     end if
  end do
end program test

File: libgomp/testsuite/libgomp.oacc-c-c++-common/pr70828.c
/* Subarray declared on enclosing data construct.  */

#include <assert.h>

int
main ()
{
  int a[100], i;

  for (i = 0; i < 100; i++)
    a[i] = 0;

#pragma acc data copy(a[10:80])
  {
    #pragma acc parallel loop
    for (i = 10; i < 90; i++)
      a[i] = i;
  }

  for (i = 0; i < 100; i++)
    if (i >= 10 && i < 90)
      assert (a[i] == i);
    else
      assert (a[i] == 0);

  return 0;
}

/* { dg-xfail-run-if "PR70828" { ! openacc_host_selected } } */
